### DISCUSSION TITLE
trying to compile simple stdio app for Plus/4 and C128 and configs not good? c64 default config is okay

### DISCUSSION THREAD

**Discussion by apws** [cc65/cc65#2052] [Q&A] [answered]
Hi, I am trying to compile simple stdio application and using standard c64 cc65 config its okay, its working.
But when I try ANY Plus/4 or C128 config, the same simple approach as for c64 fails. What can be wrong?

cc65.exe ..\..\src\VMEX.c -o VMEX.s
cl65 -C c64.cfg ..\..\src\VMEX.c -o VMEX.bin

**Comment by polluks** [comment, marked as answer]
Did you switch the target with `-t`?

**Reply by apws** [reply]
dont think so... the c64 example works as above, so I MUST also give -t and config/target name? is it enough to support stdio app in single .c file with several .h files? I thought I have bad configs and dont understad well cc65/ld65 (thats true...)

**Reply by apws** [reply]
and actually, its this code ... https://github.com/apws/230326-VMEX feel free to try it (license is for now as-is, this changes)

**Reply by apws** [reply]
it is compiling/running as latest dev-cpp project, as mplab x for various 8/16 bit MCUs, as cc65 for c64, in Segger Studio for RISC-V

**Reply by apws** [reply]
thank you, it helped - simply no -C but -t and it is okay, again thanks!!

**Comment by polluks** [comment]
It's simple: Choose your target, done.
If the target's default config does not fit, also choose a special config.
By the way your code seems not be ready for 16 bit int.

**Reply by apws** [reply]
yes, it is not :-) tests are failing ... have you any hints for C newbie? its even harder, that microchip XC8 doesnt fail for CLTI test for example, and I now dont know if even test itself isnt bad, too simple - in 32bit its okay, in XC8 its failing on multiply routines, expected, ... such testing on 8bit is the reason I am here, and on cc65 its failing even more .... I expect something around typecasting, sign-extension etc .... will see

**Reply by polluks** [reply]
Indeed, it's an advanced topic and cc65 is some kind of least common denominator. The [Rationale](https://www.open-std.org/jtc1/sc22/wg14/www/C99RationaleV5.10.pdf) is a good reference. I suppose some tests are too strict, you have to use long instead of int, etc.

**Reply by apws** [reply]
ya, thanks, I am using typedefs around stdint for better readability for me, dont like the C type names much... and for cc65 I already had modified math multiply routines to use custom structs as cc65 cant 64bit types, but even 32bit multiply is failing, but also on XC8, I initially dont tested this well in XC8/8bit AVRs too, as tests was monolythic - now I can split compilation of test to smaller chunks to fit into MCU flash and can in debug see the results - AND that multiply routines was in fact "designed by GPT4", I was glad for solution without 64bit types, but apparently... its only machine, as we are only humans :-))

//typedef uint64_t  TU64; //not cc65/8bits - muldiv solved in 32bits with custom TU64X !!!
//typedef int64_t   TS64; //not cc65/8bits - muldiv solved in 32bits with custom TS64X !!!
typedef uint32_t  TU32;
typedef int32_t   TS32;
typedef uint16_t  TU16;
typedef int16_t   TS16;
typedef uint8_t   TU8;
typedef int8_t    TS8;
//typedef bool      TBOOL;
typedef char      TCHAR; //??

**Reply by polluks** [reply]
I see. NB cc65's char is unsigned, see https://stackoverflow.com/a/40772217/1430535
BTW GPT3 is very weak in 6502, it was always creating new mnemonics :-)

**Reply by apws** [reply]
yeah, sometimes its very funny, but pushes man to think more and GPT4 quite understands even my new designed assembler abstraction )). its good colleague

**Reply by apws** [reply]
thanks for your support, all implemented tests are now OK down to cc65; multiply was okay, it was small bug in ADDI ... next will be z88dk and CP/M :-)
